test/videojs.ads.snapshot.js

```diff
@@ -1,467 +1,365 @@
+  this.player.trigger('loadstart');
+  this.player.trigger('loadedmetadata');
+  this.player.ads.endLinearAdMode();
+
+  // resume playback
+  this.player.src('http://media.w3.org/2010/05/sintel/trailer.mp4');
+  this.player.trigger('loadstart');
+  this.player.trigger('canplay');
+  this.clock.tick(1000);
+  assert.strictEqual(playSpy.callCount, 1, 'content playback resumed');
+  this.player.trigger('playing');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 1, 'A content-playback event should have been triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'playing', 'The reason for content-playback should have been playing');
 
   // if the video ends (regardless of burned in post-roll or otherwise) when
   // stopLinearAdMode fires next we should not hit play() since we have reached
   // the end of the stream
-  playCalled = false;
-  player.ended = function() {
+  this.player.ended = function() {
     return true;
   };
-  player.trigger('ended');
-  //trigger a post-roll
-  player.ads.startLinearAdMode();
-  player.src('//example.com/ad.mp4');
-  player.trigger('loadstart');
-  player.trigger('loadedmetadata');
-  player.ads.endLinearAdMode();
-  player.trigger('playing');
-  player.trigger('ended');
-
-  equal(player.ads.state, 'content-playback', 'Player should be in content-playback state after a post-roll');
-  ok(!playCalled, 'content playback should not have been resumed');
-  equal(contentPlaybackFired, 2, 'A content-playback event should have been triggered');
-  equal(contentPlaybackReason, 'ended', 'The reason for content-playback should have been ended');
-});
-
-test('snapshot does not resume playback after a burned-in post-roll', function() {
-  var
-    playCalled = false,
-    loadCalled = false;
 
-  player.trigger('adsready');
-  player.trigger('play');
-
-  // spy on relevant player methods
-  player.play = function() {
-    playCalled = true;
-  };
+  this.player.trigger('ended');
+  playSpy.reset();
+
+  // trigger a post-roll
+  this.player.ads.startLinearAdMode();
+  this.player.src('//example.com/ad.mp4');
+  this.player.trigger('loadstart');
+  this.player.trigger('loadedmetadata');
+  this.player.ads.endLinearAdMode();
+  this.player.trigger('playing');
+  this.player.trigger('ended');
+  assert.strictEqual(this.player.ads.state, 'content-playback', 'Player should be in content-playback state after a post-roll');
+  assert.strictEqual(playSpy.callCount, 0, 'content playback should not have been resumed');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 2, 'A content-playback event should have been triggered');
+  assert.strictEqual(this.contentPlaybackReason(1), 'ended', 'The reason for content-playback should have been ended');
+});
 
-  player.load = function() {
-    loadCalled = true;
-  };
+QUnit.test('snapshot does not resume playback after a burned-in post-roll', function(assert) {
+  assert.expect(9);
+  this.player.trigger('adsready');
+  this.player.tech.trigger('play');
```

Comment:
Going to note here that this is because triggering "play" on the player does not cause anything to happen. Rather, these events bubble up from the tech.